### PR TITLE
Implement pre-commit hooks.

### PR DESCRIPTION
**Pull request by owenlittlejohns**
## Description

This PR adds automated checking to all commits via pre-commit hooks. Currently it checks the following things:

* Trailing whitespace
* Blank lines at the end of the file.
* Valid JSON and YAML syntax.
* `ruff` linting.
* `black` code formatting

Why haven't I added `mypy`? So far all these checks have left the service code untouched. `mypy` will descend down dependencies trees into a whole bunch of modules, and so will test things even if there weren't directly changed. Fun for a future day. (The checks implemented now will really only look at updated files - but `black` will probably have a lot to say on the next few genuine code changes...)

## Jira Issue ID

N/A - this is an IP-ish bit of fun.

## Local Test Steps

* Pull this branch.
* Follow the new instructions in the README, excluding the optional `pre-commit run --all-files`. (Exclusion is because it will try to run `black` on _everything_)
* To test this, make some minor change, use `git add` then `git commit`. You should see the checks all run when you are trying to use `git commit`.

## PR Acceptance Checklist
* ~~Jira ticket acceptance criteria met.~~
* ~~`CHANGELOG.md` updated to include high level summary of PR changes.~~
* ~~`VERSION` updated if publishing a release.~~
* [x] Tests added/updated and passing.
* [x] Documentation updated (if needed).

### CODE REVIEW

File: tests/unit/test_generate_umm_var.py [hunk 1]
```diff
@@ -1,9 +1,6 @@
-from os.path import exists
-from shutil import copy, rmtree
-from tempfile import TemporaryDirectory
+from shutil import copy
 from unittest import TestCase
 from unittest.mock import ANY, patch
-import json
```
Comment (owenlittlejohns) [inline tests/unit/test_generate_umm_var.py:6]: This is `ruff` removing unused imports in this test module. I ran the tests locally before pushing and they still passed. The CI/CD for this PR also looks good, so I'm comfortable with this change.

File: tests/test_code_format.py [hunk 2]
```diff
@@ -5,27 +5,29 @@
...
 
     def test_pycodestyle_adherence(self):
-        """ Ensure all code in the `varinfo` directory adheres to PEP8 defined
-            standards.
+        """Ensure all code in the `varinfo` directory adheres to PEP8 defined
+        standards.
 
         """
-        style_guide = StyleGuide(ignore=['E501', 'W503'])
+        style_guide = StyleGuide(ignore=['E501', 'W503', 'E203', 'E701'])
```
Comment (owenlittlejohns) [inline tests/test_code_format.py:31]: [See here for reference](https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html#pycodestyle)

File: README.md [hunk 3]
```diff
@@ -206,6 +206,39 @@ Run `unittest` suite:
...
+* JSON files have valid formats.
+* [ruff](https://github.com/astral-sh/ruff) Python linting checks.
+* [black](https://black.readthedocs.io/en/stable/index.html) Python code
+  formatting checks.
+
+To enable these checks:
+
+```bash
+# Install pre-commit Python package as part of test requirements:
+pip install -r tests/pip_test_requirements.txt
```
Comment (flamingbear) [inline README.md:225]:
```suggestion
pip install -r dev-requirements.txt
```

**Review by flamingbear — APPROVED** [review]
Looks good, one suggestion that needs to be made, but I can approve.  🙌

**Pull request merged**